src/folderlistmodel.cpp

```diff
@@ -333,6 +339,68 @@ int FolderListModel::standardFolderIndex(FolderStandardType folderType)
     return -1;
 }
 
+bool FolderListModel::lessThan(const QMailFolderId &idA, const QMailFolderId &idB)
+{
+    if (!idA.isValid()) {
```

Comment:
These isValid() tests will break StrictWeakOrdering, I would change them to Q_ASSERTs.